[CORE-737] Do not refresh submissions history when viewing all submissions

samanehsan opened the pull request:
### Jira Ticket: https://broadworkbench.atlassian.net/browse/CORE-737

<!-- ### Dependencies --->
<!-- Include any dependent tickets and describe the relationship. Include any relevant Jira tickets. --->

## Summary of changes:
<!--Please give an abridged version of the ticket description here and/or fill out the following fields.-->

### What
The UI does not refresh running submissions if all submissions are selected.

### Why
When a user viewed all submissions, the refresh was still scheduled with the original "30 days" filter so if there were any running submissions the page would refresh to only show the last 30 days (while "all submissions" was still selected/displayed in the drop-down).

### Testing strategy
<!-- Note that changes impacting components in Storybook stories can be viewed at
https://www.chromatic.com/library?appId=65fc89c9335768720ff8605a&branch=<branch>
The branch corresponding to this PR is selected, and changes can be reviewed by commit. --->

While running the UI locally, I ran a submission and confirmed that the submission history does not refresh when "All Submissions" is selected, but it still refreshes when "30 days" is selected.

<!-- ### Visual Aids -->
<!-- https://support.apple.com/guide/quicktime-player/record-your-screen-qtp97b08e666/mac -->

samanehsan committed:
Do not refresh submissions when viewing all

When a user viewed all submissions, the refresh
was scheduled with the original "30 days" filter
so if there were any running submissions the page
would refresh to only show the last 30 days
(while "all" was still selected/displayed).

Now, the UI does not refresh running submissions if
all submissions are selected.

davidangb commented on src/pages/workspaces/workspace/SubmissionHistory.js:
```diff
@@ -216,7 +216,10 @@ export const SubmissionHistory = _.flow(
+        if (scheduledRefresh.current) {
+          clearTimeout(scheduledRefresh.current);
+        }
         scheduledRefresh.current = setTimeout(refresh, 1000 * 60);
```
I _think_ there's a race condition here, but I haven't tried to prove this:
1. User has a running submission and is viewing last 30 days. This `refresh` method schedules the next `refresh` to execute in 60 seconds.
2. Before those 60 seconds are up, the user switches to view all submissions. Because `dateRange === 'all'` at this point, we don't call `clearTimeout`.
3. After 60 seconds are up, the previously-scheduled refresh executes, which populates the screen with 30 days of submissions.

I think you may need to move the `clearTimeout` higher in this function. In fact, I'd suggest doing it as the first-ish thing in `refresh`; it shouldn't need to wait until after the ajax call to `listSubmissions`.

samanehsan replied:
Ah good catch! I was able to reproduce it locally 😞 I fixed it by always clearing the timeout first if it is set. Then if any of the fetched submissions have an in-progress status and the dateRange is not 'all', set a new refresh timer.

samanehsan committed:
Fix refresh timer race condition

samanehsan committed:
Remove refresh on mount

The submissions will already refresh when the dateRange
gets initially set, and whenever that value changes.

davidangb reviewed (approved):
lgtm, thanks!
